cmake/Doctest.cmake

```diff
@@ -1,6 +1,11 @@
 if(ENABLE_DOCTESTS)
-    add_definitions(-DENABLE_DOCTEST_IN_LIBRARY)
```

Comment:
This line still needs to be here, otherwise, all the tests in headers will not be built. Running the test executable in your branch just runs one test. It should run several. Putting back this one line should solve that.